make --always-make deletes some files

This issue was quite a bit fun to debug!

Turns out VSCode recommends installing the [Makefile tools](https://github.com/Microsoft/vscode-makefile-tools/) extension, which runs this command-line to get some info on the project:

```bash
make --dry-run --always-make --keep-going --print-directory
```

This command-line on an EPICS top removes all `CONFIGS` files during the installation phase. From my debugging, the files are removed by the `installEpics.pl`. Auditd says this command removed the `configure/CONFIG` file:

```bash
perl -CSD /nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/bin/linux-x86_64/installEpics.pl -d -m 444 ../CONFIG ../.
```

And `installEpics.pl` says:

```
installEpics.pl: Copy failed: No such file or directory
installEpics.pl:        ../CONFIG -> ../../configure/TEMP.CONFIG.1267659
```

<details>
<summary>Full make logs</summary>

```
make: Entering directory '/tmp/tmp.69OuGuGda6'
make -C ./configure install
make[1]: Entering directory '/tmp/tmp.69OuGuGda6/configure'
perl -CSD /nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/bin/linux-x86_64/makeMakefile.pl O.linux-x86_64 ../..
mkdir -p O.Common
make -C O.linux-x86_64 -f ../Makefile TOP=../.. \
    T_A=linux-x86_64 install
make[2]: Entering directory '/tmp/tmp.69OuGuGda6/configure/O.linux-x86_64'
Installing config file ../../configure/CONFIG
installEpics.pl: Copy failed: No such file or directory
installEpics.pl:        ../CONFIG -> ../../configure/TEMP.CONFIG.1267659
make[2]: *** [/nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/configure/RULES_BUILD:539: ../../configure/CONFIG] Error 2
Installing config file ../../configure/RELEASE
installEpics.pl: Copy failed: No such file or directory
installEpics.pl:        ../RELEASE -> ../../configure/TEMP.RELEASE.1267661
make[2]: *** [/nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/configure/RULES_BUILD:539: ../../configure/RELEASE] Error 2
Installing config file ../../configure/CONFIG_SITE
installEpics.pl: Copy failed: No such file or directory
installEpics.pl:        ../CONFIG_SITE -> ../../configure/TEMP.CONFIG_SITE.1267663
make[2]: *** [/nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/configure/RULES_BUILD:539: ../../configure/CONFIG_SITE] Error 2
Installing config file ../../configure/RULES
installEpics.pl: Copy failed: No such file or directory
installEpics.pl:        ../RULES -> ../../configure/TEMP.RULES.1267665
make[2]: *** [/nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/configure/RULES_BUILD:539: ../../configure/RULES] Error 2
../Makefile:3: Failed to remake makefile '../../configure/CONFIG'.
make[2]: Leaving directory '/tmp/tmp.69OuGuGda6/configure/O.linux-x86_64'
../../configure/CONFIG:9: Failed to remake makefile '../../configure/RELEASE'.
../../configure/CONFIG:33: Failed to remake makefile '../../configure/CONFIG_SITE'.
../Makefile:8: Failed to remake makefile '../../configure/RULES'.
make[1]: *** [/nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/configure/RULES_ARCHS:58: install.linux-x86_64] Error 2
make[1]: Leaving directory '/tmp/tmp.69OuGuGda6/configure'
make: *** [/nix/store/3jdcplbpdj2177csm7nbw3prjlaq1bdn-epics-base-7.0.7/configure/RULES_DIRS:85: configure.install] Error 2
make: Leaving directory '/tmp/tmp.69OuGuGda6'
```

</details>

Confirmed, in my case `--always-make` deleted `RULES_COMMON`:

```
ctldev% make --always-make
perl -CSD ../src/tools/makeMakefile.pl O.linux-x86_64 ../..
mkdir -p O.Common
make -C O.linux-x86_64 -f ../Makefile TOP=../.. \
    T_A=linux-x86_64 install
make[1]: Entering directory '/home/phoebus4/ANJ/epics/base/7.0/configure/O.linux-x86_64'
Installing config file ../../configure/RULES_COMMON
installEpics.pl: Copy failed: No such file or directory
installEpics.pl:	../RULES_COMMON -> ../../configure/TEMP.RULES_COMMON.3366234
make[1]: *** [../../configure/RULES_BUILD:539: ../../configure/RULES_COMMON] Error 2
make[1]: Leaving directory '/home/phoebus4/ANJ/epics/base/7.0/configure/O.linux-x86_64'
make: *** [../configure/RULES_ARCHS:58: install.linux-x86_64] Error 2
ctldev% git st .
On branch 7.0
Your branch is up to date with 'upstream/7.0'.

Changes not staged for commit:
  (use "git add/rm <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
	deleted:    RULES_COMMON

no changes added to commit (use "git add" and/or "git commit -a")
```

Interesting, I suspect this may be related to the code that copies the `configure/` tree when you have `INSTALL_LOCATION` set. I don't think it's likely to get fixed any time soon...

--- a/configure/RULES_BUILD
+++ b/configure/RULES_BUILD
@@ -532,11 +532,11 @@ endif # LOADABLE_SHRLIB_SUFFIX
 ifneq ($(INSTALL_CONFIGS),)
 $(INSTALL_CONFIG)/%: %
 	$(ECHO) "Installing config file $@"
-	@$(INSTALL) -d -m $(INSTALL_PERMISSIONS) $< $(@D)
+	@$(INSTALL) -d -m $(INSTALL_PERMISSIONS) $(abspath $< $(@D))
 
 $(INSTALL_CONFIG)/%: ../%
 	$(ECHO) "Installing config file $@"
-	@$(INSTALL) -d -m $(INSTALL_PERMISSIONS) $< $(@D)
+	@$(INSTALL) -d -m $(INSTALL_PERMISSIONS) $(abspath $< $(@D))
 endif
 
 $(INSTALL_INCLUDE)/%: $(COMMON_DIR)/%

--- a/src/tools/installEpics.pl
+++ b/src/tools/installEpics.pl
@@ -57,6 +57,9 @@ foreach my $source (@ARGV) {
     my $temp   = "$install_dir/TEMP.$name.$$";
     my $target = "$install_dir/$name";
 
+    # Don't try to install the file if it already exists
+    next if $source eq $target;
+
     if (-f $target) {
         next if -M $target < -M $source and -C $target < -C $source;
         # Remove old target, making sure it is deletable first

